Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelUploadTests.swift

```diff
@@ -0,0 +1,130 @@
+import NetworkService
+import XCTest
+
+final class SitterProfileViewModelUploadTests: XCTestCase {
+    private var viewModel: SitterProfileViewModel!
+
+    override func setUp() {
+        super.setUp()
+        URLProtocol.registerClass(MockURLProtocol.self)
+        viewModel = SitterProfileViewModel()
+    }
+
+    override func tearDown() {
+        URLProtocol.unregisterClass(MockURLProtocol.self)
+        viewModel = nil
+        super.tearDown()
+    }
+
+    func testUploadSuccessfullyExecuted() async {
+        MockURLProtocol.requestHandler = { request in
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: XCTUnwrap(request.url),
+                    statusCode: 200,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, nil)
+        }
+
+        do {
+            try await viewModel.upload()
+            XCTAssertEqual(viewModel.isErrorOccurred, false)
+        } catch {}
+    }
+
+    func testBadRequestStatusCodeInResponseTriggersError() async {
+        MockURLProtocol.requestHandler = { request in
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: XCTUnwrap(request.url),
+                    statusCode: 400,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, nil)
+        }
+
+        do {
+            try await viewModel.upload()
+            XCTFail("Expected failure, but got success")
+        } catch {
+            XCTAssertEqual(viewModel.isErrorOccurred, true)
+        }
+    }
+
+    func testRandomErrorStatusCodeInResponseTriggersError() async {
+        let randomStatusCode = Int.random(in: 400...499)
```

Comment:
Do you think using an unpredictable case is a good idea?
What is the goal of the test, if you tested all cases in the next test?

Reply:
Yes you are right, deleted with 06c4f69